query_modules/text_search_module.cpp

```diff
@@ -42,16 +51,90 @@ void TextSearch::Search(mgp_list *args, mgp_graph *memgraph_graph, mgp_result *r
   }
 }
 
+void TextSearch::RegexSearch(mgp_list *args, mgp_graph *memgraph_graph, mgp_result *result, mgp_memory *memory) {
+  mgp::MemoryDispatcherGuard guard{memory};
+  const auto record_factory = mgp::RecordFactory(result);
+  auto arguments = mgp::List(args);
+
+  try {
+    const auto *index_name = arguments[0].ValueString().data();
+    const auto *search_query = arguments[1].ValueString().data();
+    for (const auto &node : mgp::SearchTextIndex(memgraph_graph, index_name, search_query, "regex")) {
+      auto record = record_factory.NewRecord();
+      record.Insert(TextSearch::kReturnNode.data(), node.ValueNode());
+    }
+  } catch (const std::exception &e) {
```

Comment:
Can we be more specific with the exceptions we catch?

Reply:
This is general by design (catch _any_ exception and set its error message as result), compare any query module in mage.